src/models/differentiable/evotree_model.jl

```diff
@@ -0,0 +1,147 @@
+using DataFrames
+using SliceMap
+using EvoTrees
+using MLJBase
+
+"""
+This type provides a basic interface to gradient-boosted tree models from the MLJ library.
+However, this is not be the final version of the interface: full support for EvoTrees has not been implemented yet and the `logits` and `probs` methods will be changed in the process of doing that if needed.
+"""
+
+"""
+    EvoTreeModel <: AbstractMLJModel
+
+Constructor for gradient-boosted decision trees from the EvoTrees.jl library. 
+"""
+struct EvoTreeModel <: AbstractDifferentiableModel
+    model::Any
+    likelihood::Symbol
+    function EvoTreeModel(model, likelihood)
+        if likelihood ∈ [:classification_binary, :classification_multi]
+            new(model, likelihood)
+        else
+            throw(
+                ArgumentError(
+                    "`type` should be in `[:classification_binary, :classification_multi]1.
+                    Support for regressors has not been implemented yet.`"
+                ),
+            )
+        end
+    end
+end
+
+"""
+Outer constructor method for EvoTreeModel.
+"""
+function EvoTreeModel(model::Any; likelihood::Symbol=:classification_binary)
+    return EvoTreeModel(model, likelihood)
+end
+
+# Methods
+"""
+    logits(M::MLJModel, X::AbstractArray)
+
+Calculates the logit scores output by the model M for the input data X.
+
+# Arguments
+- `M::MLJModel`: The model selected by the user. Must be a model from the MLJ library.
+- `X::AbstractArray`: The feature vector for which the logit scores are calculated.
+
+# Returns
+- `logits::Matrix`: A matrix of logits for each output class for each data point in X.
+
+# Example
+logits = Models.logits(M, x) # calculates the logit scores for each output class for the data point x
+"""
+function logits(M::EvoTreeModel, X::AbstractArray)
+    p = probs(M, X)
+    if M.likelihood == :classification_binary
+        logits = log.(p ./ (1 .- p))
+    else
+        logits = log.(p)
+    end
+    return logits
+end
+
+"""
+    probs(M::EvoTreeModel, X::AbstractArray{<:Number, 2})
+
+Calculates the probability scores for each output class for the two-dimensional input data matrix X.
+
+# Arguments
+- `M::EvoTreeModel`: The EvoTree model.
+- `X::AbstractArray`: The feature vector for which the predictions are made.
+
+# Returns
+- `p::Matrix`: A matrix of probability scores for each output class for each data point in X.
+
+# Example
+probabilities = Models.probs(M, X) # calculates the probability scores for each output class for each data point in X.
+"""
+function probs(M::EvoTreeModel, X::AbstractArray{<:Number,2})
+    output = MLJBase.predict(M.model, DataFrame(X', :auto))
+    p = pdf(output, classes(output))'
```

Comment:
Are you sure you need to use this `'` operation? It's an adjoint operation, not a transpose (`transpose()` in Julia) (it caused a few errors on my end when working with PyTorch models)

Reply:
Good, lmk what are the results